kedro/extras/datasets/spark/databricks_deltatable_dataset.py

```diff
@@ -0,0 +1,88 @@
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Databricks specific DataSets"""
+import functools
+import logging
+from typing import Any, Dict
+
+from delta.tables import DeltaTable
+from pyspark.sql import DataFrame, DataFrameWriter
+
+from kedro.extras.datasets.spark import SparkDataSet
+from kedro.io.core import DataSetError, NodeStatus, Version
+
+logger = logging.getLogger(__name__)
+
+
+class DeltaTableDataset(SparkDataSet):
+    """
+    Schema validation: https://docs.databricks.com/delta/delta-batch.html#schema-validation-1
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        filepath: str,
+        delta_options: Dict[str, Any] = None,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+        version: Version = None,
+        credentials: Dict[str, Any] = None,
+    ) -> None:
+        super().__init__(
+            filepath=filepath,
+            file_format="delta",
+            load_args=load_args,
+            save_args=save_args,
+            version=version,
+            credentials=credentials,
+        )
+        self._delta_options = delta_options
+
+    def _add_options(self, df: DataFrame) -> DataFrameWriter:
+        # DeltaTable specific opts, such as `schemaValidation`
+        if self._delta_options:
+            df_writer = df.write
+            return functools.reduce(
+                lambda dfw, opt: df_writer.option(opt, self._delta_options[opt]),
+                self._delta_options,
+                df_writer,
+            )
+        return df.write
+
+    def _load(self):
+        load_path = self._fs_prefix + str(self._get_load_path())
+        return DeltaTable.forPath(self._get_spark(), load_path)
+
+    def _save(self, io: NodeStatus):
+        if io == NodeStatus.SUCCESS:
+            logger.info(
+                "`NodeStatus` returned SUCCESS, `save` operation was performed"
+                "within the context of the node."
+            )
+        else:
+            raise DataSetError("`NodeStatus` returned something other than SUCCESS")
```

Comment:
I've been wondering if we could piggyback on the `confirm` method concept that we use in `PartitionedDataSet`. In `PartitionedDataSet` it just updates the checkpoint so that we don't reprocess the same partitions. I think conceptually that's closer than a `save`, and it means the node itself doesn't have to return some special thing. But on the flip side I think it does mean that it's not captured in Kedro Viz as an output..

Reply:
I like this thought

Reply:
I think that is right - let me try the functionality and how it behaves with the dataset!